lib/puppet/configurer.rb

```diff
@@ -73,6 +74,10 @@ def retrieve_catalog(query_options)
     if (Puppet[:use_cached_catalog] && result = retrieve_catalog_from_cache(query_options))
       @cached_catalog_status = 'explicitly_requested'
 
+      Puppet.info _("Using cached catalog from environment '%{environment}'") % { environment: result.environment }
+    elsif (@pluginsync_failed && result = retrieve_catalog_from_cache(query_options))
+      @cached_catalog_status = 'on_pluginsync_failure'
```

Comment:
Should we log that pluginsync failed? Especially since `Puppet[:ignore_plugin_errors] = false` and the user might care about pluginsync failing.

Reply:
This was already checked when variable was set to true:

https://github.com/puppetlabs/puppet/blob/cf2f609c1b470a9a0e285ae3c33181d216d044b1/lib/puppet/configurer.rb#L539-L540